packages/core/admin/webpack.config.js

```diff
@@ -49,6 +49,14 @@ module.exports = ({
       ]
     : [];
 
+  /**
+   * converts the full path to just the plugin path
+   * e.g. `/Users/username/strapi/node_modules/@scope/plugin-name`
+   * to `@scope/plugin-name`
+   */
+  const tsxPlugins = pluginsPath.map((dir) => dir.split('node_modules/')[1]).join('|');
+  const excludeRegex = new RegExp(`node_modules/(?!(${tsxPlugins}))`);
```

Comment:
Does that work on Windows, or do we have to use e.g. [`path.sep`](https://nodejs.org/docs/latest/api/path.html#pathsep)?

Reply:
excellent question. I have no idea. Do we have anyone with windows? 👀

Reply:
> Building complex regexp from external data can be very dangerous because paths can contain special regexp characters. It can also lead to bugs when using other platforms or package managers.

I don't think this would lead to a bug, this simply gets the name of the plugins. The issue with remove `node_modules` as you keep suggesting is that it increases the memory footprint significantly and therefore I don't believe it's a viable option.

Reply:
> The issue with remove node_modules as you keep suggesting is that it increases the memory footprint significantly and therefore I don't believe it's a viable option.

But I'm not suggesting this now, I created two new commits, that are still ignoring `node_modules` as previously, what do you mean by "keep suggesting"? Did you checked new changes?

> I don't think this would lead to a bug, this simply gets the name of the plugins.

Any arguments for this? Are you sure it will work on all platforms - windows, mac, linux? Will it work on `npm`, `yarn` and `pnpm` as well?

Reply:
> Did you checked new changes?

I did, you're suggesting the loader should look like this:

```ts
        {
          test: /\.tsx?$/,
          loader: require.resolve('esbuild-loader'),
          oneOf: [
            {
              include: pluginsPath,
            },
            {
              include: [cacheDir],
              exclude: /node_modules/,
            },
          ],
          options: {
            loader: 'tsx',
            target: 'es2015',
          },
        },
```

but you're not excluding `node_modules` when using the `pluginsPath`. Previously in the memory test above, removing this increases the amount of memory used. However, I will look into memory profiling this to see it's effects.

> Are you sure it will work on all platforms - windows, mac, linux? Will it work on npm, yarn and pnpm as well?

As from the first comment, we're looking into it's usage on different platforms. In regards to the package managers, i've used all three, and they populate a `node_modules` folder, so I do think it would work. Alternatively we get the list of plugins without their respective paths anyway so we could just use that function I believe.

Reply:
I believe that in my previous suggestion memory increase was caused by `cacheDir`, not by `pluginsPath`. At this moment by using `oneOf`, basically I'm saying that webpack should use `tsx` compiler for `cacheDir` without `node_modules`, but when it comes to `pluginsPath` it should force use `tsx` compiler.
In your PR you're doing the same thing, but by removing `pluginsPath` from exclude.

> However, I will look into memory profiling this to see it's effects.

it would be great, i am curious about the results as well.

Reply:
I've ran the test 6 times, the majority of which were consuming ~780mb of memory which then put it nearly as high as the first test I ran above.

Reply:
@joshuaellis Could you please check one more time but for:
```js
{
  test: /\.tsx?$/,
  loader: require.resolve('esbuild-loader'),
  oneOf: [
    {
      include: pluginsPath,
      exclude: /node_modules\/@strapi/
    },
    {
      include: [cacheDir],
      exclude: /node_modules/,
    },
  ],
  options: {
    loader: 'tsx',
    target: 'es2015',
  },
},
```
How can I contact you? Are you on strapi community on discord?